### PR TITLE
[CF-424] Improve error reporting

### PR DESCRIPTION
**Pull request by antonf**
This patch add error log messages that will help to find error in
configuration.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
👍

**Comment by roman-verchikov** [comment]
+1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/discover.py [hunk 1]
```diff
@@ -176,6 +177,7 @@ def discover_all(cfg, cloud):
         LOG.debug('Finished discovering %s using %s',
```
Comment (archyufa) [inline cloudferry/discover.py]: Can we add comment that: " Results of discovery can be found in migration_data.db" ?

File: cloudferry/config.py [hunk 2]
```diff
@@ -232,6 +232,8 @@ class SshGateway(ConfigSection):
     password = fields.String(missing=None)
     private_key = fields.String(missing=None)
     gateway = fields.Nested('self', missing=None)
+    connection_attempts = fields.Integer(missing=1)
```
Comment (archyufa) [inline cloudferry/config.py]: Can you add comment what is the parameter does?

File: cloudferry/config.py [hunk 3]
```diff
@@ -232,6 +232,8 @@ class SshGateway(ConfigSection):
...
+    attempt_failure_sleep = fields.Float(missing=10.0)
```
Comment (archyufa) [inline cloudferry/config.py]: Can you add comment what is the parameter does?

File: cloudferry/discover.py (hunk 1 above)
Comment (antonf) [inline cloudferry/discover.py]: But you can't find any "results" there. This file is not designed to be viewed by human being.

File: cloudferry/config.py (hunk 2 above)
Comment (antonf) [inline cloudferry/config.py]: No, I can't. I still need to figure out how to generate configuration documentation from config, so It is separate task.

File: cloudferry/config.py (hunk 3 above)
Comment (antonf) [inline cloudferry/config.py]: No, I can't. I still need to figure out how to generate configuration documentation from config, so It is separate task.

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
Can be merged

**Pull request merged**